app/containers/ItunesContainer/index.js

```diff
@@ -4,47 +4,152 @@
  *
  */
 
-import React, { memo } from 'react';
+import React, { memo, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { injectIntl } from 'react-intl';
-import { Input } from 'antd';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
 import { injectSaga } from 'redux-injectors';
-import { Input } from 'antd';
-import makeSelectItunesContainer from './selectors';
+import { Input, Skeleton } from 'antd';
+import { debounce, get, isEmpty } from 'lodash';
+import styled from 'styled-components';
 import saga from './saga';
+import ItunesCard from '@components/ItunesCard/index';
 import { itunesContainerCreators } from './reducer';
-import { debounce, isEmpty } from 'lodash';
+import { selectItuneName, selectItunesData, selectItunesError } from './selectors';
+import For from '@components/For/index';
+import If from '@components/If/index';
+import { T } from '@components/T/index';
+import * as colors from '@themes/colors';
 
 const { Search } = Input;
 
-export function ItunesContainer({ dispatchItunesData, dispatchClearItunesData }) {
+const FlexContainer = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+`;
+
+const Container = styled.div`
+  && {
+    max-width: 75rem;
+    width: 100%;
+    margin: 0 auto;
+    padding: 1rem;
+    color: ${colors.error};
+  }
+`;
+
+export function ItunesContainer({ dispatchItunesData, dispatchClearItunesData, itunesData, itunesError, ituneName }) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(itunesData, 'results', itunesError);
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [itunesData, itunesError]);
+
+  useEffect(() => {
+    if (ituneName && !itunesData?.results?.length) {
+      dispatchItunesData(ituneName);
+      setLoading(true);
+    }
+  }, []);
+
   const handleOnChange = (ituneName) => {
-    if (!isEmpty(ituneName)) {
+    if (ituneName) {
       dispatchItunesData(ituneName);
+      setLoading(true);
     } else {
       dispatchClearItunesData();
     }
   };
 
+  const [currTrack, setCurrTrack] = useState();
+
+  const handleOnActionClick = (audioElem) => {
+    if (!audioElem.paused) {
+      if (!isEmpty(currTrack) && currTrack !== audioElem) {
+        currTrack.pause();
+      }
+      setCurrTrack(audioElem);
+    }
+  };
+
   const debouncedHandleOnChange = debounce(handleOnChange, 200);
 
+  const renderItunesData = () => (
+    <Skeleton loading={loading} active>
+      <For
+        of={itunesData?.results}
+        ParentComponent={FlexContainer}
+        renderItem={(item, index) => <ItunesCard key={index} itune={item} handleOnActionClick={handleOnActionClick} />}
```

Comment:
Please check this naming convention for handlers. https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
By that pattern, ItunesCard should take in onAction as a prop.

Reply:
Thanks for the reference. Updated the name.